resources/rule.rb

```diff
@@ -17,40 +17,62 @@
 
 property :rule_name, String, name_property: true
 property :description, String, default: 'Firewall rule'
-property :localip, String
-property :localport, String
-property :remoteip, String
-property :remoteport, String
-property :dir, Symbol, default: :in, equal_to: [:in, :out]
+property :local_address, String
+property :local_port, String
+property :remote_address, String
+property :remote_port, String
+property :direction, Symbol, default: :inbound, equal_to: [:inbound, :outbound],
+          coerce: proc { |d| d.is_a?(String) ? d.downcase.to_sym : d }
 property :protocol, String, default: 'TCP'
-property :firewall_action, Symbol, default: :allow, equal_to: [:allow, :block, :bypass]
-property :profile, Symbol, default: :any, equal_to: [:public, :private, :domain, :any]
+property :firewall_action, Symbol, default: :allow, equal_to: [:allow, :block, :notconfigured],
+          coerce: proc { |f| f.is_a?(String) ? f.downcase.to_sym : f }
+property :profile, Symbol, default: :any, equal_to: [:public, :private, :domain, :any, :notapplicable],
+          coerce: proc { |p| p.is_a?(String) ? p.downcase.to_sym : p }
 property :program, String
 property :service, String
-property :interfacetype, Symbol, default: :any, equal_to: [:any, :wireless, :lan, :ras]
+property :interface_type, Symbol, default: :any, equal_to: [:any, :wireless, :wired, :remoteaccess],
```

Comment:
If you want to allow strings and then coerce them to symbols you need to include String as an allowed type